src/page/class.rs

```diff
@@ -1,16 +1,33 @@
 // SPDX-License-Identifier: Apache-2.0
 
-/// The class of a page
-///
-/// This enumeration identifies the kind of one or more pages. This type is
-/// named `PAGE_TYPE` in the Intel documentation.
+use crate::page::{Flags, SecInfo};
+
+/// The type of an enclave page (see Intel SDM Volume 3D section 34.12.2).
+/// Enclave Page Cache Map (EPCM) hols this information for each valid enclave
+/// page.
 #[repr(u8)]
 #[non_exhaustive]
 #[derive(Copy, Clone, Debug, PartialEq)]
 pub enum Class {
+    /// SGX Enclave Control Structure (SECS)
     Secs = 0,
+    /// Thread Control Structure (TCS)
     Tcs = 1,
-    Reg = 2,
-    Va = 3,
-    Trim = 4,
+    /// Regularular page
+    Regular = 2,
+    /// Version Array (VA) page
+    VersionArray = 3,
+    /// Removable from a running enclave
+    Trimmed = 4,
+    /// The first page of a shadow stack
+    ShadowStackFirst = 5,
+    /// A shadow stack page
+    ShadowStackRest = 6,
+}
+
+impl Class {
+    /// Convert to SecInfo with the given flags.
+    pub fn info(&self, flags: impl Into<Option<Flags>>) -> SecInfo {
```

Comment:
@npmccallum Can we just call this `secinfo` for clarity's sake?

Reply:
Or actually, strictly following coding conventions, it should be `sec_info`, which looks kind of ugly. Since we've taken `page` out of flags and class, maybe actually better idea would be to rename `SecInfo` to `Info`. That would be at least consistent. By all practical means and purposes SecInfo is PageInfo.

Reply:
Or if we go into this, I would actually even go to propose to call it `Descriptor`, which is I think the name hat documents its best. It's a page descriptor, right? It's mapping to SDM can be resolved in the struct documentation comment.

Reply:
We did anyway that kind of rename by renaming type to class so I think this would very sound naming:
* `page::Class`
* `page::Flags`
* `page::Descriptor`
What you think?

Reply:
I think for comprehensibility it is better to stay closer to the SDM terminology unless there is a clear reason to avoid it.

Reply:
Can cope with that.